### PR TITLE
Add general doc guidelines to Community section

### PR DESCRIPTION
**Pull request by chenette**
This commit includes a new "Doc Guidelines" section with all planned guidelines (to minimize merge conflicts). Some links will be broken until the other guideline PRs have been merged.

Signed-off-by: Anne Chenette <chenette@bitwise.io>

### CODE REVIEW

File: community/doc_guidelines/general.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  Licensed under Creative Commons Attribution 4.0 International License
+  https://creativecommons.org/licenses/by/4.0/
+-->
+
+* Be consistent.
+
+* Follow the [Splinter capitalization guidelines](capitalization.md).
+
+* For punctuation and font changes, follow [Google's typographic
+  conventions](https://developers.google.com/style/typographical-conventions).
```
Comment (agunde406) [inline community/doc_guidelines/general.md:14]: looks like this link redirects to https://developers.google.com/style/text-formatting any reason not to link directly?

Comment (chenette) [reply community/doc_guidelines/general.md]: Updated this link and others.  

I also made a few other improvements while I was there: added an intro, reorganized for clarity, fixed items that don't follow the doc guidelines, and added an attribution for the Google Developer Documentation Style Guide (per their request).

File: community/doc_guidelines/general.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  https://creativecommons.org/licenses/by/4.0/
+-->
+
+The Splinter documentation guidelines are a work in progress.
+For items not covered here, see
+[plainlanguage.gov](https://www.plainlanguage.gov/guidelines/),
+the [Merriam-Webster's online dictionary](https://www.merriam-webster.com/),
+and the [Google developer documentation style
+guide](https://developers.google.com/style).
+Many thanks to the Google writers who provide this clear, comprehensive guide.
```
Comment (ltseeley) [inline community/doc_guidelines/general.md:15]: "this clear, comprehensive guide" sounds like you're referring to this guide (the one you're writing). Might want to reword slightly. :)

Comment (ltseeley) [reply community/doc_guidelines/general.md:15]: Didn't realize most of the content on this page is from that guide; seems fine on second read.

**Comment by ltseeley** [comment]
Commit was cherry-picked in #141, closing

**Pull request closed without merging**